### PR TITLE
Use controller-gen to generate the CRD

### PR DESCRIPTION
**Pull request by bhatpmk**
This PR addresses the extended validation in Kubernetes v1.18.10, which causes the error mentioned in VZ-1747 during the installation of Verrazzano.

Based on the suggestion and sample Makefile change provided by Bob, I have modified the CRD generation in verrazzano-crd-generator and verrazzano-helidon-app-operator, to address the issue mentioned in VZ-1747 during the installation of Verrazzano with k8s 1.18.10 cluster. The change here avoids the workaround to update the generated CRDs, done by earlier PRs.

The generated CRD contains a null value for time stamp - creationTimestamp: null, which did not create any new failures in acceptance test

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -38,3 +42,12 @@ go-mod:
 	chmod +x vendor/${CODEGEN_PATH}/generate-groups.sh
 	cp -R ${GOPATH}/pkg/mod/${CODEGEN_PATH}@${codeGenVer}/cmd/defaulter-gen vendor/${CODEGEN_PATH}/cmd/defaulter-gen
 	chmod -R +w vendor/${CODEGEN_PATH}/cmd/defaulter-gen
+
++.PHONY: controller-gen
+controller-gen:
+	go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.4.1
```
Comment (markxnelson) [inline Makefile:48]: looks like you are introducing a new dependency here

File: THIRD_PARTY_LICENSES.txt [hunk 2]
```diff
@@ -1556,4 +1556,4 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 
 ATTRIBUTION-HELPER-GENERATED:
-License file based on go.mod with md5 sum: b4906c2e76a90616b28a4f2631e53bc9
+License file based on go.mod with md5 sum: 97792ee2b8c2014f3904ef303f3a96c8
```
Comment (markxnelson) [inline THIRD_PARTY_LICENSES.txt:1559]: looks odd - if you changed go.mod we expect to see more change here

File: go.mod [hunk 3]
```diff
@@ -4,6 +4,7 @@ require (
 	github.com/operator-framework/operator-sdk v0.18.1
 	github.com/spf13/pflag v1.0.5
 	github.com/stretchr/testify v1.5.1
+	golang.org/x/tools v0.0.0-20200616195046-dc31b401abb5 // indirect
```
Comment (markxnelson) [inline go.mod:7]: go mod tidy

File: Makefile (hunk 1 above)
Comment (bhatpmk) [reply Makefile:48]: I had a workaround for https://github.com/kubernetes-sigs/controller-tools/issues/529, where I was using sed to insert the default value for the generated CRD - https://github.com/verrazzano/verrazzano-crd-generator/pull/16

Bob suggested controller-gen to generate the CRD, which sets the default for protocol, which is a better option than processing the generated CRD. So we have a new dependency here

File: THIRD_PARTY_LICENSES.txt (hunk 2 above)
Comment (bhatpmk) [reply THIRD_PARTY_LICENSES.txt:1559]: This is my fault, I will fix this

Comment (bhatpmk) [reply THIRD_PARTY_LICENSES.txt:1559]: @markxnelson I have followed the instruction to install the attribution-helper and run "attribution-helper gen". I don't see any other change in THIRD_PARTY_LICENSES.txt. Am I missing something ?

File: go.mod (hunk 3 above)
Comment (bhatpmk) [reply go.mod:7]: I can see go mod tidy is already part of the Makefile, I don't see any change if I run go mod tidy from the shell.

Comment (markxnelson) [reply go.mod:7]: ok, thanks for checking

**Comment by bhatpmk** [comment]
I have a clean acceptance test using OKE Cluster v1.18.10 and the test source from a branch -
https://build.verrazzano.io/job/verrazzano-magic-dns-acceptance-tests/job/pabhat%252FVZ-1747/10/

**Review by mgianatagh — APPROVED** [review]
I tested the build and attribution-helper gen. They worked fine.

LGTM assuming approval for controller-gen@v0.4.1

**Review by bobdonat — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (markxnelson) [reply Makefile:48]: i will process the new dependency approval, but please refer to decision log,  regarding the use of code generators in makefiles

Comment (bhatpmk) [reply Makefile:48]: Thank you @markxnelson . I was not aware of the decision log, sorry for that. I will generate the code in a temporary directory and update the README to copy the generated artefacts to the appropriate location.

Comment (mgianatagh) [reply Makefile:48]: Generating the code in a temporary directory and adding manual steps does not sound right to me.  I thought the intention of the code generation decision was to not have code generation targets be a build dependency.  Instead they need to intentionally be run when you know you made a change that requires newly generated code.

Also, I am not sure that applies to this repository.  I think that any change you make in the verrazzano-crd-generator repository is to generate updated code.  That is the nature of this repo.  All the actions of the build target are to generate code.

Comment (bhatpmk) [reply Makefile:48]: @markxnelson With my latest change, the usage of code generators in Makefile is as per the decision log.

**Review by mgianatagh — APPROVED** [review]

**Pull request merged**